.travis.yml

```diff
@@ -21,11 +21,14 @@ before_install:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]] && [[ "$TRAVIS_BRANCH$TRAVIS_EVENT_TYPE" == "masterpush" ]]; then brew update; fi
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]] && [[ "$TRAVIS_BRANCH$TRAVIS_EVENT_TYPE" == "masterpush" ]]; then brew install arm-none-eabi-gcc; fi
 
-before_script: (cargo install rustfmt || true)
+before_script:
+  - (cargo install rustfmt || true)
+  - if [[ "$TRAVIS_OS_NAME" != "osx" ]] || [[ "$TRAVIS_BRANCH$TRAVIS_EVENT_TYPE" == "masterpush" ]]; then npm install -g markdown-toc; fi
```

Comment:
I don't disagree. But, it doesn't seem that easy to cache global npm packages (see [an example job](https://travis-ci.org/helena-project/tock/jobs/216606471#L197) and [so1](http://stackoverflow.com/questions/30240237/how-do-i-cache-global-npm-packages-on-travis-ci) and [so2](http://stackoverflow.com/questions/31419940/cache-global-node-modules-on-travis-in-the-correct-directory)). Also it costs 3 seconds on linux and 0.01 seconds on mac.

I continue to get the feeling that Travis might be the single hardest to use easy-to-use thing ever created.